onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -241,8 +244,18 @@ def collect_data(self, data_reader: CalibrationDataReader):
             if not inputs:
                 break
             self.intermediate_outputs.append(self.infer_session.run(None, inputs))
-
-        if len(self.intermediate_outputs) == 0:
+            if (
+                self.max_intermediate_outputs is not None
+                and len(self.intermediate_outputs) == self.max_intermediate_outputs
+            ):
+                self.compute_range()
+                self.clear_collected_data()
+
+        if (
+            self.max_intermediate_outputs is None
```

Comment:
There is no need to check "self.max_intermediate_outputs is None"

Reply:
I think you are right yes, as `self.calibrate_tensors_range is None` is sufficient.